### PR TITLE
Selection screens (EXPOSUREAPP-7465)

### PR DESCRIPTION
**Pull request by jurajkusnier**
- [x] WIP
- [x] #3548 has to be merged at first

### CODE REVIEW

**Review by NataliaLemmerth — DISMISSED** [review]
UA approved

**Review by kolyaopahle — DISMISSED** [review]
LGTM 👍 i think we need some more modifications to location storage to allow for state/district storage and retrieval, but i think we should do it in its own PR

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import timber.log.Timber
+
+class FederalStateSelectionViewModel @AssistedInject constructor(
+    @Assisted val selectedFederalStateShortName: String?,
+    @AppContext private val context: Context,
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt:29]: can you please suppress the warning here ?

**Review by mtwalli — COMMENTED** [review]
Add new item card sometimes is scrolled and sometimes not , is this a known issue?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    )
+                }
+                .sortedBy { it.label.get(context).lowercase() }
+        }
+
+    val listItems: LiveData<List<ListItem>> = if (selectedFederalStateShortName != null) {
+        districtSource
+    } else {
+        federalStateSource
+    }.catch { Timber.e(it, "Error sourcing list items.") }
```
[the remark this answers is not in the document]
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt:65]: This implementation is very similar to what we already have in `AnalyticsUserInputViewModel` so I'm using the same error handling as there.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionFragment.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
+
+class FederalStateSelectionFragment : Fragment(R.layout.federal_state_selection_fragment), AutoInject {
+
+    val navArgs by navArgs<FederalStateSelectionFragmentArgs>()
```
[the remark this answers is not in the document]
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionFragment.kt]: updated

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionFragment.kt [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as FederalStateSelectionViewModel.Factory
+            factory.create(navArgs.selectedFederalStateShortName)
+        }
+    )
+
+    private val binding: FederalStateSelectionFragmentBinding by viewBinding()
+
+    val itemAdapter = FederalStateAdapter { viewModel.selectUserInfoItem(it) }
```
[the remark this answers is not in the document]
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionFragment.kt]: updated

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        .asLiveData(context = dispatcherProvider.Default)
```
[the remark this answers is not in the document]
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt]: updated

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt (hunk 1 above)
Comment (jurajkusnier) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt:29]: updated

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/statistics/ui/stateselection/FederalStateSelectionViewModel.kt:64]: Ok

**Comment by jurajkusnier** [comment]
> Add new item card sometimes is scrolled and sometimes not , is this a known issue?

Yes, it will be handled in another PR as a bugfix

**Review by kolyaopahle — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**